### PR TITLE
Adding ConstraintLayout solution to the project

### PR DESCRIPTION
**Pull request by yombunker**
- Updating the version of the ConstraintLayout library
- Adding layout that uses ConstraintLayout as the main solver for the problem
- Updating the MainActivity to add the code needed in order to animate the changes

### CODE REVIEW

File: app/src/main/java/com/elyeproj/horizontaltextarragement/MainActivity.kt [hunk 1]
```diff
@@ -30,5 +32,12 @@ class MainActivity : AppCompatActivity() {
                 space_worst_wrap.visibility = View.VISIBLE
             }
         }
+
+        btn_constraint_remove.setOnClickListener {
+            val isVisible = txt_worst_constraint.visibility == View.VISIBLE
+            TransitionManager.beginDelayedTransition(constraint_layout)
```
Comment (elye) [inline app/src/main/java/com/elyeproj/horizontaltextarragement/MainActivity.kt:38]: This is not required, once we have `animateLayoutChanges="true"` in the ConstraitLayout.

File: app/src/main/res/layout/layout_constraint_space.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<android.support.constraint.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:id="@+id/constraint_layout"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="#0ff">
```
Comment (elye) [inline app/src/main/res/layout/layout_constraint_space.xml:7]: Propose add `animateLayoutChanges="true"` in ConstraintLayout.

**Review by elye — COMMENTED** [review]
Do consider remove the Kotlin animation code, and add animateLayoutChanges in the XML.

File: app/build.gradle [hunk 3]
```diff
@@ -28,6 +28,7 @@ dependencies {
     })
     compile 'com.android.support:appcompat-v7:25.4.0'
     compile 'com.android.support.constraint:constraint-layout:1.0.0-beta4'
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
```
Comment (elye) [inline app/build.gradle:31]: You might want to remove the `compile 'com.android.support.constraint:constraint-layout:1.0.0-beta4'`

**Review by elye — CHANGES_REQUESTED** [review]
Propose several changes.

File: app/build.gradle (hunk 3 above)
Comment (yombunker) [reply app/build.gradle:31]: sorry forgot to delete that line :P

**Pull request merged**